### ISSUE TITLE
Home assistant service call for groups from HA bridge device

### ISSUE THREAD

**Issue by papa-legba** [bwssytems/ha-bridge#444]
First HA bridge is amazing. Really stable and I've got all my lights hue and lutron dimmers working great. 

Is there a way I can set up a home assistant device on the bridge and have it call a specific domain/service/json data in home assistant. 

Basic is controlling light groups created in home assistant using the homeassistant domain.. Or adjusting the hdmi_cec domain to change my tv. Or using the lock domain to call specific door lock/unlock. 

All examples I've found online relate to earlier versions of the bridge so I don't which fields to fill out with the headers and the on url ect.

**Comment by bwssytems** [comment]
Which version are you using currently?

**Comment by quadflight** [comment]
If you are using the current version of HABRIDGE all lights,switches,scripts and automations are accessible by default. Already tested scritps, lights and switches. Works great. 

![image](https://cloud.githubusercontent.com/assets/24735733/22630292/3d53f1c0-ebac-11e6-86e1-fd91a6abef87.png)

You can adjust the domain/service from the `target item`

**Comment by papa-legba** [comment]
I'm using 4.1.1. 

Again I'm sure the issue is my lack of experience configuring home assistant. 

If I wanted to create a bridge element to turn on a group of lights as identified in home assistant how do I build this element. 

In home assistant I can create the results by using the developer tools/ domain:home assistant/service: turn on/then the json info that maps to the group. Basically I want to know how to map the information I test in home assistant to the device I'm building in the bridge. 

Again I know this is probably obvious to the more experienced but I'm about two weeks into this.. 

Thanks.

**Comment by bwssytems** [comment]
You should be able to use the Home Assistant tab to build your device. There should not be any need to do manual editing of the calls from the build item. So, you can create the group/scene in HomeAssistant, or you could build a "group" by selecting a device then build, then go back to the HomeAssistant screen and select the next device and then hit build. This will be additive to the device and when you are done, you can hit the add bridge device.

**Comment by papa-legba** [comment]
I can verify that lights and scripts and switches work. Also I'm sure automation a also work but I haven't set those up. 

But If for example I build an element using a group defined in home assistant that the bridge doesn't turn the group off or on. 

{"entityId":"group.main_lights","hassName":"Home Assistant","state":"on"}

Or if I build a lock. 

{"entityId":"lock.front_door","hassName":"Home Assistant","state":"on"}

It also doesn't turn on or off. 
I believe this is because groups or locks aren't controlled by the light domain but rather the homeassistant domain or lock domain. Again I'm new to this. Just trying to figure what to tweak in the target ID to trigger the right domain/service/(data, json) info. 

In a nutshell if I can fire it from the home assistant developer tab I'm trying to understand how to directly map those commands to the target ID field. 

Again the HA bridge is amazing just trying to see if a noob can get it to do more.

**Comment by cscattaglia917** [comment]
I can verify that I see the same behavior as papa-legba, in regards to controlling HASS groups.
I see this in the home-assistant log after I attempt to control a group via HA Bridge:
`17-02-11 11:44:34 homeassistant.core: Unable to find service group/turn_off
17-02-11 11:44:57 homeassistant.core: Unable to find service group/turn_on`

When I manually run JSON from within HomeAssistant's "Services" tab, I use the light domain, with turn_on/turn_off as the service, and pass the group through as well using: {"entity_id":"group.all_lights"}
And that works for me.

**Comment by bwssytems** [comment]
I'll take a look at the groups.

**Comment by bwssytems** [comment]
So, it looks like groups are only available for control through the dashboard. There is no service entry to operate on them from the HomeAssistant RestfulAPI

**Comment by cscattaglia917** [comment]
You can control groups(and AFAIK any other domain that isn't explicitly listed in the docs) with the homeassistant domain.
Ex:)
I am able to POST to https://domain.com/api/services/homeassistant/turn_on to control my groups. 

Hope this helps! :+1:

**Comment by bwssytems** [comment]
Alright, I see that under the homeassistant domain. I'll need to do a special check for group when the command fires.

**Comment by bwssytems** [comment]
Well, that works much better.... fixed in 4.2.1

**Comment by cscattaglia917** [comment]
You're awesome, thank you so much!
